### PR TITLE
feat(components): TextField

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #5850

Adds a TextField to Phoenix that's more aligned with aria components composition. Also has 2 sizes.

Screenshot 2024-12-30 at 4 05 27 PM

### CODE REVIEW

File: app/src/components/textfield/TextField.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { css } from "@emotion/react";
+
+import { fieldBaseCSS } from "@phoenix/components/field/styles";
+import { SizingProps } from "@phoenix/components/types";
+export interface TextFieldProps extends AriaTextFieldProps, SizingProps {}
+
+const textFieldCSS = css`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
```
Comment (mikeldking) [inline app/src/components/textfield/TextField.tsx:15]: I'm not quite happy with this - I think it should be fit-content but not sure how to make form layouts nice quite yet.

File: app/src/components/field/styles.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    &::placeholder {
+      color: var(--ac-global-text-color-300);
+      font-style: italic;
+    }
+  }
+  [slot="description"],
+  .react-aria-FieldError {
+    /* The overriding cascade here is non ideal but it lets us have only one notion of text  */
+    font-size: var(--ac-global-font-size-xs) !important;
```
Comment (mikeldking) [inline app/src/components/field/styles.ts:47]: This is really not ideal but I wanted to make the `Text` field composible like aria components. Problem is that in this nested context I want the font smaller...

File: app/src/components/index.tsx [hunk 3]
```diff
@@ -1,3 +1,11 @@
+import type {
+  FieldErrorProps,
+  InputProps,
+  LabelProps,
+} from "react-aria-components";
+// eslint-disable-next-line no-duplicate-imports
+import { FieldError, Input, Label } from "react-aria-components";
```
Comment (cephalization) [inline app/src/components/index.tsx:7]: Can you remove the eslint override by changing this to `export { FieldError, Input, Label } from 'react-aria-components'`

File: app/src/components/textfield/TextField.tsx (hunk 1 above)
Comment (cephalization) [reply app/src/components/textfield/TextField.tsx:15]: I tend to like 100% myself, and then control the width in the parent container via flex with siblings or grid cols

**Review by cephalization — APPROVED** [review]

File: app/src/components/textfield/TextField.tsx (hunk 1 above)
Comment (mikeldking) [reply app/src/components/textfield/TextField.tsx:15]: I think you're right - that tends to be what I want.

File: app/src/components/index.tsx (hunk 3 above)
Comment (mikeldking) [reply app/src/components/index.tsx:7]: Oh yeah that's much cleaner

**Pull request merged**